src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -133,7 +137,10 @@ void SoxSink::write(audio::Frame& frame) {
     write_(buffer_data, buffer_pos);
 }
 
-bool SoxSink::prepare_() {
+bool SoxSink::setup_buffer_() {
+    size_t requested_device_rate = sample_rate();
```

Comment:
*requested rate* (i.e. the rate we're asking to be used for device) is stored in `out_signal_.rate`. And `sample_rate()` returns `output_->signal.rate`, which is *actual rate* selected for device. And we're using actual rate to calculate buffer size.

So we're correctly using `sample_rate()` here, but the variable is named incorrectly.

I suggest to rename the variable to `rate` or `actual_rate` or something like that, or just remove it at all:

```
buffer_size_ = packet::ns_to_size(frame_length_, sample_rate(), channels_);
```

Reply:
done.